src/lib/text-utils.ts

```diff
@@ -30,6 +30,9 @@ interface TrimmedTextResult {
 // Trim text, placing an ellipsis in the middle, with a slight bias towards
 // keeping text from the beginning rather than the end
 export function buildTrimmedText(text: string, length: number): TrimmedTextResult {
+  // make thread name more concise by spliting full path and showing the last part as thread name
+  const textFullPath = text.split("/")
+  text = textFullPath[textFullPath.length - 1]
```

Comment:
Considering that speedscope accepts not only Golang style pprof data, but also other kind of profiling data from other languages, this might not be a general solution I think.

Reply:
Yes, you'r right. I will go through all the supported languages and find a more general and better display way.